File: PacketBuilder.vb
Public Class PacketBuilder
    Public Property Bytes As New List(Of Byte)
    Public Sub Write(ByVal Value As Byte)
        Bytes.Add(Value)
    End Sub
    Public Sub Write(ByVal Values As Byte())
        For Each v As Byte In Values
            Write(v)
        Next
    End Sub
    Public Sub Write(ByVal Value As SByte)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As Short)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As Integer)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As Long)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As UShort)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As UInteger)
        Write(Reverse(BitConverter.GetBytes(Value)))
    End Sub
    Public Sub Write(ByVal Value As ULong)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Value As Boolean)
        Write(BitConverter.GetBytes(Value))
    End Sub
    Public Sub Write(ByVal Text As String)
        If Text.Length = 0 Or Text = "" Then Bytes.Add(0) : Exit Sub
        Dim aarray = System.Text.Encoding.UTF8.GetBytes(Text)
        Dim barray = {CByte(Text.Length)}
        Bytes.AddRange(Combine({barray, aarray}))
    End Sub
    Private Function Reverse(ByVal Bytes As Byte()) As Byte()
        Dim l As List(Of Byte) = BasicUFL.ToList(Bytes)
        l.Reverse()
        Return l.ToArray
    End Function

    Public Function GetBytes() As Byte()
        Return Bytes.ToArray
    End Function
End Class

File: PacketBuilder.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class PacketBuilder
{
	public List<byte> Bytes { get; set; }
	public void Write(byte Value)
	{
		Bytes.Add(Value);
	}
	public void Write(byte[] Values)
	{
		foreach (byte v in Values) {
			Write(v);
		}
	}
	public void Write(sbyte Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(short Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(int Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(long Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(ushort Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(uint Value)
	{
		Write(Reverse(BitConverter.GetBytes(Value)));
	}
	public void Write(ulong Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(bool Value)
	{
		Write(BitConverter.GetBytes(Value));
	}
	public void Write(string Text)
	{
		if (Text.Length == 0 | string.IsNullOrEmpty(Text)){Bytes.Add(0);return;
}
		byte[] aarray = System.Text.Encoding.UTF8.GetBytes(Text);
		byte[] barray = { Convert.ToByte(Text.Length) };
        byte[][] carray = { barray, aarray };
        Bytes.AddRange(Misc.Combine(carray));
	}
	private byte[] Reverse(byte[] Bytes)
	{
        List<byte> l = new List<byte>(Bytes);
		l.Reverse();
		return l.ToArray();
	}

	public byte[] GetBytes()
	{
		return Bytes.ToArray();
	}
}
